asserts/database.go

```diff
@@ -171,14 +175,19 @@ var (
 	fingerprintLike = regexp.MustCompile("^[0-9a-f]*$")
 )
 
-// PublicKey exports the public part of a stored key pair for identity
-// by matching the given fingerprint suffix, it is an error if no or more
-// than one key pair is found.
-func (db *Database) PublicKey(authorityID string, fingerprintSuffix string) (PublicKey, error) {
-	if !fingerprintLike.MatchString(fingerprintSuffix) {
-		return nil, fmt.Errorf("fingerprint suffix contains unexpected chars: %q", fingerprintSuffix)
+func (db *Database) safeGetPrivateKey(authorityID, keyID string) (PrivateKey, error) {
+	if keyID == "" {
+		return nil, fmt.Errorf("key id is empty")
 	}
-	privKey, err := db.keypairMgr.Find(authorityID, fingerprintSuffix)
+	if !fingerprintLike.MatchString(keyID) {
+		return nil, fmt.Errorf("key id contains unexpected chars: %q", keyID)
+	}
+	return db.keypairMgr.Get(authorityID, keyID)
+}
+
+// PublicKey exports the public part of a stored key pair for identity and key id.
```

Comment:
How about:

`// PublicKey returns the public key owned by authorityID that has the given key id.`